src/JustSaying/Messaging/MessageSerialization/NewtonsoftSerializer.cs

```diff
@@ -52,6 +56,35 @@ public string Serialize(Message message, bool serializeForSnsPublishing, string
             return JsonConvert.SerializeObject(context, _settings);
         }
 
+        public MessageAttributes GetMessageAttributes(string message)
+        {
+            var props = JObject.Parse(message).Value<JObject>("MessageAttributes")?.Properties();
+            if (props == null) return new MessageAttributes();
+            var dict = new Dictionary<string, MessageAttributeValue>();
+
+            foreach (var prop in props)
+            {
+                var propData = prop.Value as JObject;
+                if (propData == null) continue;
+
+                var dataType = propData.GetValue("Type", StringComparison.Ordinal);
+                if (dataType == null) continue;
+
+                var isString = dataType.ToString().Equals("String", StringComparison.Ordinal);
```

Comment:
Doesn't Netwonsoft expose a property (`ValueKind`, `Kind` or similar) that tells you this with an enum? Similarly I think you could use it instead of using `as JObject`.

Reply:
I've reworked this to use the `Type` field - makes things a fair bit simpler. Will get the integration tests running with goaws then we can see if it actually works 😄